arrow-fx-coroutines-test/src/main/kotlin/arrow/fx/coroutines/stream/test/TestStream.kt

```diff
@@ -0,0 +1,81 @@
+package arrow.fx.coroutines.stream.test
+
+import arrow.core.getOrElse
+import arrow.core.orElse
+import arrow.fx.coroutines.stream.Stream
+import arrow.fx.coroutines.stream.concurrent.Queue
+import arrow.fx.coroutines.stream.drain
+import arrow.fx.coroutines.stream.firstOrNull
+import arrow.fx.coroutines.stream.toList
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import java.util.concurrent.TimeUnit
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.ExperimentalTime
+import kotlin.time.toDuration
+import arrow.fx.coroutines.Duration as ArrowDuration
+
+@ExperimentalTime
+fun testStream(
+  timeout: Duration = 1.toDuration(DurationUnit.SECONDS),
+  block: suspend TestStream.() -> Unit
+): Unit = testStreamCompat(timeout = timeout.toArrowDuration(), block = block)
+
+fun testStreamCompat(
+  timeout: ArrowDuration = ArrowDuration(1, TimeUnit.SECONDS),
+  block: suspend TestStream.() -> Unit
+): Unit = runBlocking { TestStream(timeout = timeout).block() }
+
+// TODO: Convert to Kotlin Duration when not experimental
+class TestStream internal constructor(private val timeout: ArrowDuration) {
+
+  fun Stream<*>.capture() {
+    GlobalScope.launch {
+      runCatching { through(queue.enqueue()).drain() }
+        .onFailure { exceptionQueue.enqueue1(it) }
+    }
```

Comment:
```suggestion
    arrow.fx.coroutines.Enviroment().unsafeRunAsnyc {
      through(queue.enqueue())
        .handleErrorWith { Stream.effect { exceptionQueue.enqueue1(it) } }
        .drain()
    }
```

Reply:
Hmm, this one makes the "expect items in order" test hang and not timeout or success

Reply:
Damn it, I typed Sync instead of Async... in it goes :)